client/components/GaugeComponent.js

```diff
@@ -0,0 +1,156 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+const canvasStyle = {border: '1px solid #000000'};
+
+export default class GaugeComponent extends React.Component {
+    constructor(props) {
+        super(props);
+    }
+
+    componentDidMount() {
+        this.defineClassConstants();
+    }
+
+    componentWillUpdate() {
+        this.context.clearRect(0, 0, this.canvas.width, this.canvas.height);
+        this.drawGauge();
+    }
+
+    drawGauge() {
+        for (let i = this.xAxisBuffer; i <= this.rightSideEnd; i++) {
+            this.optionallyPlaceMarkers(i)
```

Comment:
here, it's better to draw 5 markers explicitly than iterate over all pixels and then draw only 5 of them